File: domain/src/main/java/de/x4fyr/timeline/adapter/TodoListAdapter.java
package de.x4fyr.timeline.adapter;

import de.x4fyr.timeline.domain.elements.TodoElement;

/**
 * @author Benedikt Volkmer
 *         Created on 11/14/16.
 */
public interface TodoListAdapter {

    TodoElement saveToToDoList(TodoElement element);

    void deleteFromTodoList(TodoElement element);
}

File: domain/src/main/kotlin/de/x4fyr/timeline/adapter/TodoListAdapter.kt
package de.x4fyr.timeline.adapter

import de.x4fyr.timeline.domain.elements.TodoElement
import org.joda.time.LocalDate

/**
 * @author Benedikt Volkmer
 *         Created on 11/14/16.
 */
interface TodoListAdapter {

    fun saveToToDoList(element: TodoElement): TodoElement

    fun deleteFromTodoList(element: TodoElement)

    fun getAll(): Set<TodoElement>

    fun  getByDate(date: LocalDate): Set<TodoElement>
}
